client/session.go

```diff
@@ -702,6 +714,48 @@ func (s *session) Truncate(namespace string) (int64, error) {
 	return truncated, resultErr.FinalError()
 }
 
+func (s *session) peersForShard(shard uint32) ([]hostQueue, error) {
+	s.RLock()
+	peers := make([]hostQueue, 0, s.topoMap.Replicas())
```

Comment:
Does replicas include the origin host? If so, should this be peers := make([]hostQueue, 0, s.topoMap.Replicas()-1)?